src/app/core/store/videos/videos.action.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Lumeer: Modern Data Definition and Processing Platform
+ *
+ * Copyright (C) since 2017 Answer Institute, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {VideoModel} from './video.model';
+import {Action} from '@ngrx/store';
+
+export enum VideosActionType {
+  CLEAR_VIDEOS = '[Videos] Clear Videos',
+  LOAD_VIDEOS = '[Videos] Load Videos',
+  LOAD_VIDEOS_FAILURE = '[Videos] Load Video Failure',
+  REGISTER_VIDEOS = '[Videos] Register Video',
+}
+
+export namespace VideosAction {
+  export class ClearVideos implements Action {
+    public readonly type = VideosActionType.CLEAR_VIDEOS;
+
+    public constructor() {}
+  }
+
+  export class LoadVideos implements Action {
+    public readonly type = VideosActionType.LOAD_VIDEOS;
+
+    public constructor(public payload: {videos: {[id: string]: number}; apiKey: string}) {}
+  }
+
+  export class LoadVideosFailure implements Action {
+    public readonly type = VideosActionType.LOAD_VIDEOS_FAILURE;
+
+    public constructor(public payload: {error: any}) {}
+  }
+
+  export class RegisterVideos implements Action {
+    public readonly type = VideosActionType.REGISTER_VIDEOS;
+
+    public constructor(public payload: VideoModel[]) {}
```

Comment:
This does not follow action payload structure used everywhere else in the application.

Reply:
To be honest, I never noticed any payload structure. Can you tell me the secret please? What would you expect here? `payload: {videos: VideoModel[]}`?

Reply:
Yes, you have already used it in all other actions above.